pkg/liqoctl/network/cluster.go

```diff
@@ -211,6 +213,25 @@ func (c *Cluster) GetGatewayClient(ctx context.Context, name string) (*networkin
 	return gwClient, nil
 }
 
+func endpointHasChanged(endpoint *networkingv1alpha1.Endpoint, service *corev1.Service) bool {
+	if endpoint.ServiceType != service.Spec.Type {
+		return true
+	}
+	if len(service.Spec.Ports) > 0 {
+		if endpoint.Port != service.Spec.Ports[0].Port {
+			return true
+		}
+
+		if endpoint.NodePort != nil && *endpoint.NodePort != service.Spec.Ports[0].NodePort {
+			return true
+		}
+	}
+	if endpoint.LoadBalancerIP != nil && *endpoint.LoadBalancerIP != service.Spec.LoadBalancerIP {
+		return true
+	}
+	return false
+}
```

Comment:
What about using something like this?

https://github.com/kubernetes-sigs/controller-runtime/blob/5f8d96b043ef8ccadc544535415395be336630e8/pkg/controller/controllerutil/controllerutil.go#L216C1-L218C3

Reply:
I don't think we could use it in this case as the two types are different from each other, resulting always in a false equivalency.